### DISCUSSION TITLE
Any roadmap to potentially contributing to this project in a significant way, from a Software Engineer starting point?

### DISCUSSION THREAD

**Discussion by AZMCode** [koka-lang/koka#651] [General]
Hi!

I'm... Well this is pseudonymous but suffice to say my initials are in my username so no big secret either to those who know me.

I'm a young software engineer from Bolivia. Mostly self taught, but I am going through my local university's Software Engineering degree. This is mostly because it is impractical for me to travel abroad and get a proper CS education.

Due to this I don't appear to have many opportunities to study the theory of CS outside of wrangling myself some relatively useful information from the vast Interwebs.

As I'm sure many a CS(ish) student with OCD has done in their early years, I've looked around for my perfect language to code in. And honestly, after searching the landscape of both FP and OOP and even Stack-based languages, (And a couple years of flirting with pure Haskell, which I still find quite neat, and explains my long leave of abscence from this community), I've come to the personal and completely subjective belief that Algebraic Effects are one of the most interesting abstractions I've ever seen, they should definitely be built into the language if at all possible, and that Koka in particular does it best to my liking.  Even if not as powerful as Monad Transformers or similar, the composability benefits and separation of implementation and definition of an effect IMO largely outstrip any sheen Monads initially had for me.

As such, I'd love to eventually be able to contribute meaningfully to the compiler internals, help develop language features, etc.

Eventually.

Emphasis on Eventually.

Like many years from now.

Regardless of whether I end up actually following through on that goal, I've been headed in that general direction for years now, and even if it doesn't end up working out work-wise, this is where I want to go personally.

Problem is, I'm directionless. Mostly consuming whatever little theory comes my way through the internet, and no idea how to build a roadmap to being fully educated and practiced to the degree needed to contribute to language compiler projects like this one in a deep and meaningful way.

Any pointers as to what I can do/who to search for/what to look out for in order to fulfill these goals?

I come to this community in particular due to how welcoming it has been to me, and how close it is to my particular goals.

I appreciate any and all constructive advice. Cheerio!

**Comment by osa1** [comment]
> As such, I'd love to eventually be able to contribute meaningfully to the compiler internals, help develop language features, etc.
>
> Eventually.
>
> Emphasis on Eventually.
>
> Like many years from now.

Why wait years? If you have the time, just pick an issue and start coding. Repeat for a few years and you will be quite proficient in compilers.
